setuptools/config/expand.py

```diff
@@ -0,0 +1,288 @@
+"""Utility functions to expand configuration directives or special values
+(such glob patterns).
+
+We can split the process of interpreting configuration files into 2 steps:
+
+1. The parsing the file contents from strings to value objects
+   that can be understand by Python (for example a string with a comma
+   separated list of keywords into an actual Python list of strings).
+
+2. The expansion (or post-processing) of these values according to the
+   semantics ``setuptools`` assign to them (for example a configuration field
+   with the ``file:`` directive should be expanded from a list of file paths to
+   a single string with the contents of those files concatenated)
+
+This module focus on the second step, and therefore allow sharing the expansion
+functions among several configuration file formats.
+"""
+import ast
+import importlib
+import io
+import os
+import sys
+from glob import iglob
+from itertools import chain
+
+from distutils.errors import DistutilsOptionError
+
+chain_iter = chain.from_iterable
+
+
+class StaticModule:
+    """
+    Attempt to load the module by the name
+    """
+
+    def __init__(self, name, spec):
+        with open(spec.origin) as strm:
+            src = strm.read()
+        module = ast.parse(src)
+        vars(self).update(locals())
+        del self.self
+
+    def __getattr__(self, attr):
+        try:
+            return next(
+                ast.literal_eval(statement.value)
+                for statement in self.module.body
+                if isinstance(statement, ast.Assign)
+                for target in statement.targets
+                if isinstance(target, ast.Name) and target.id == attr
+            )
+        except Exception as e:
+            raise AttributeError(
+                "{self.name} has no attribute {attr}".format(**locals())
+            ) from e
+
+
+def glob_relative(patterns, root_dir=None):
+    """Expand the list of glob patterns, but preserving relative paths.
+
+    :param list[str] patterns: List of glob patterns
+    :param str root_dir: Path to which globs should be relative
+                         (current directory by default)
+    :rtype: list
+    """
+    glob_characters = ('*', '?', '[', ']', '{', '}')
```

Comment:
It seems like a set could be semantically more correct here
```suggestion
    glob_characters = {'*', '?', '[', ']', '{', '}'}
```